metaflow/runner/subprocess_manager.py

```diff
@@ -10,23 +10,55 @@
 from typing import Callable, Dict, Iterator, List, Optional, Tuple
 
 
+def send_signals(pid, signal):
+    subprocess.call(["pkill", signal, "-P", str(pid)])
```

Comment:
Not using `check_call` here on purpose, if the subprocess doesn't have any descendants, `check_call` raises an exception. This (for some arcane reason) slowed the code down substantially.